src/parser.js

```diff
@@ -1343,6 +1343,14 @@ function mapForGuard(guardNode, meta) {
   return guardClause;
 }
 
+function guardLoopSource(source) {
+  if (!n.Identifier.check(source) && !n.MemberExpression.check(source)) {
```

Comment:
Right.  Maybe it flows easier if I check for `Literal`/`ObjectExpression`/`TemplateLiteral`?